### ISSUE TITLE
Alerts for user-provided value thresholds

### ISSUE THREAD

**Issue by waldyrious** [lainsce/coin#16]
It would be great if the app could be configured to pop up a system notification whenever a currency goes above (or below) a value threshold.

In case it helps, I'm thinking of a functionality similar to the alerts of the Coinbase app for Android.

ps - this could be useful as an alternative to (some use cases of) #13

**Comment by lainsce** [comment]
So you're saying that the app should have notifications when the value changes as in comparison to a user defined value? Wouldn't that be chaotic due to the 10-second update system?

**Comment by waldyrious** [comment]
That would only happen if the user picks a threshold that within the usual variability of the currency. As an example, say you'd want to know when the US Dollar is worth below 0.80 €, or above 0.90 €, or some other not so common value, which would make such alerts useful (otherwise one would have to be checking the app all the time).

Note that I'm talking about *crossing* a threshold (i.e. changing from under X to over X, not just being above X at every check). In addition, the app could disable an alert as soon as it's triggered, so that if the currency starts fluctuating near that value (which is likely) it won't trigger the alert repeatedly. This is actually how the Coinbase app does it, so there's precedent for this UX.

**Comment by lainsce** [comment]
I understood your issue now, but I don't know how:  
>In addition, the app could disable an alert as soon as it's triggered, so that if the currency starts fluctuating near that value (which is likely) it won't trigger the alert repeatedly.  

works in Vala, I'd need to research this.

**Comment by waldyrious** [comment]
That's certainly an extra for this feature. It would be nice to have, but not essential. I could open a separate issue to track that as an enhancement, if you'd like.

**Comment by lainsce** [comment]
No need to open it, since these two things are part of each other, here let's make it into a checkbox thing:

- [ ] Settings panel with Gtk.Entry to store threshold
- [ ] Alert when crossing the threshold, but only once
- [ ] Show notification when it crosses

**Comment by waldyrious** [comment]
> Settings panel with Gtk.Entry to store threshold

Just to be clear, will there be multiple thresholds definable, or just one?

**Comment by lainsce** [comment]
I can only store one threshold or one range of thresholds at a time..

**Comment by lainsce** [comment]
After much consideration, testing & trial and error, I think I can't do this because the Notifications API doesn't have a means of actually only showing the notification once then stopping, and then doing the same on subsequent openings of the app. I'm sorry, but I'm closing this bug. :cry:

**Comment by waldyrious** [comment]
No problem. Thank you for looking into it :+1:
